### PR TITLE
refactor: remove deployment plan

### PR DESCRIPTION
**Pull request by PaulFarault**
#### Which issue(s) this PR fixes

Fixes None

#### Additional comments

Remove DeploymentPlan class:

- DeploymentIterator is now based directly on a DeploymentLog and iterates over OperationLogs.
- Tests and commands have been refactored accordingly.

#### Agreements

- [X] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [X] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

**Comment by mdrutel** [comment]
When I execute ```tdp browse 1```, I have this output in the "Deployment" part :
```
              from_dag  <function DeploymentLog.from_dag at 0x7f1d227af378>
       from_operations  <function DeploymentLog.from_operations at 0x7f1d227af840>
      from_reconfigure  <function DeploymentLog.from_reconfigure at 0x7f1d227af8c8>
from_failed_deployment  <function DeploymentLog.from_failed_deployment at 0x7f1d227af950>
```

**Comment by mdrutel** [comment]
There's a reference to the ```deployment_plan``` in the docs/source/tdp.core.deployment.rst
```
tdp.core.deployment.deployment\_plan module
-------------------------------------------

.. automodule:: tdp.core.deployment.deployment_plan
   :members:
   :undoc-members:
   :show-inheritance:
```

**Comment by PaulFarault** [comment]
Force pushed after rebase on master and fix suggestion.

File: tdp/core/deployment/test_deployment_runner.py [hunk 1]
```diff
@@ -135,10 +144,10 @@ def test_service_log_not_emitted_when_config_start_wrong_order(
     minimal_collections, deployment_runner
 ):
     """deployment plan containing start then config should not emit service log"""
-    deployment_plan = DeploymentPlan.from_operations(
-        minimal_collections, ["mock_node_start", "mock_node_config"]
+    deployment_log = DeploymentLog.from_operations(
+        minimal_collections, ["mock_start", "mock_config"]
```
Comment (rpignolet) [inline tdp/core/deployment/test_deployment_runner.py:148]: You do not need to change these operations.
```suggestion
        minimal_collections, ["mock_node_start", "mock_node_config"]
```

File: tdp/core/deployment/test_deployment_runner.py [hunk 2]
```diff
@@ -168,16 +176,17 @@ def test_service_log_emitted_once_with_multiple_config_and_start_on_same_compone
...
 ):
     """deployment plan containing multiple config, start, and restart should emit only one service log"""
-    deployment_plan = DeploymentPlan.from_operations(
+    deployment_log = DeploymentLog.from_operations(
         minimal_collections,
         [
             "mock_node_config",
             "mock_node_start",
             "mock_node_config",
+            "mock_node_start",
```
Comment (rpignolet) [inline tdp/core/deployment/test_deployment_runner.py:185]: Why adding a `start` action ? It doesn't make sense to have a `start` before a `restart`

**Review by rpignolet — APPROVED** [review]

**Review by mdrutel — APPROVED** [review]

**Pull request merged**